lib/services/runtime/handlers/interaction.ts

```diff
@@ -36,50 +35,32 @@ export const InteractionHandler: HandlerFactory<Node.Interaction.Node, typeof ut
       return node.id;
     }
 
-    let nextId: string | null | undefined;
-    let variableMap: SlotMapping[] | null = null;
-
+    // request for this turn has been processed, delete request
+    runtime.turn.delete(T.REQUEST);
     const { intent } = request.payload;
 
-    // check if there is a choice in the node that fulfills intent
-    node.interactions.forEach((choice, i) => {
-      if (choice.intent && utils.formatIntentName(choice.intent) === intent.name) {
-        if (choice.goTo) {
-          runtime.turn.set(T.REQUEST, { ...request, payload: { ...request.payload, intent: { name: choice.goTo.intentName, slots: [] } } });
-        } else {
-          variableMap = choice.mappings ?? null;
-          nextId = node.nextIds[choice.nextIdIndex ?? i];
-        }
+    const index = node.interactions.findIndex((choice) => choice.intent && utils.formatIntentName(choice.intent) === intent.name);
```

Comment:
the only logic that is going to change here, is that instead of matching the LAST instance of the intent in the array, it will match the first one. I don't think this should matter because choice blocks should not have 2 of the same intents